tests/utime/test_utime.c

```diff
@@ -32,20 +33,35 @@ void test() {
   // will fail
   struct utimbuf t = {1000000000, 1000000000};
 
-  utime("writeable", &t);
+  errno = 0;
```

Comment:
Hmm .. if this line is needed.. how was the test passing before?

Reply:
I have no clue ...

Reply:
What test are you running that fails if you remove this line?

`./tests/runner core2.test_utime` works fine for on HEAD.

Reply:
I executed the test manually with ``$ emcc -s NODERAWFS -o test_utime.js test_utime.c && node test_utime.js``. The test failed with error message ``RuntimeError: Aborted(Assertion failed: !errno, at: test_utime.c,37,test)``. My guess is that some function earlier left errno != 0.

From man errno(3):

> The  value in errno is significant only when the return value of the call indicated an error (i.e., -1 from most system calls; -1 or NULL from most library functions); a function that succeeds is allowed to change errno.  The value of errno is never set to zero by any system call or library function.

In other words, if you want to use errno for error checking, then you have to reset errno to 0 before a function call and check it immediately after the function call.

Reply:
Yes, I'm only confused about how that test was passing before.

It sounds like the tests passes because its not run with NODERAWFS today?  i.e. `./tests/runner core2.test_utime` does not set `NODERAWFS`.   Perhaps we should annotate that test with `also_with_noderawfs`